djconnectwise/sync.py

```diff
@@ -1782,10 +1792,10 @@ def _assign_field_data(self, instance, json_data):
             instance.actual_end = parse(actual_end).date()
 
         try:
-            project_id = json_data['projectId']
+            project_id = json_data.get('projectId')
             related_project = models.Project.objects.get(pk=project_id)
             setattr(instance, 'project', related_project)
-        except KeyError:
+        except AttributeError:
```

Comment:
This wouldn't raise an attribute error, it would raise a DoesNotExist exception.

Reply:
Test has been done for a double check:

![image](https://user-images.githubusercontent.com/73486314/140568120-edfb498c-8677-4c58-8626-9d5a08acb68f.png)

AttributeError seems to occur.

Reply:
https://github.com/KerkhoffTechnologies/django-connectwise/pull/363#discussion_r743928219

Reply:
That's the django shell, and it looks like you imported all the models from the models package of djautotask, then you called `models`, which in this context has imported the `django.db.models` package. And that package doesn't contain the model `Project`

To test that in the shell, on `In [5]:` you should have called it as `related_project = Project.objects.get(pk=project_id)`

So, no the code in question will not raise an AttributeError

Reply:
Oh.. I overlooked the error message. You are right! Just a min.

Reply:
Thanks for pointing out this error. I appreciate your valuable review. Fixed and ready again.